File: CastIt.Domain/Dtos/AppResponseDto.cs
namespace CastIt.Domain.Dtos
{
    public class AppResponseDto<T> : EmptyResponseDto
    {
        public T Result { get; set; }

        public AppResponseDto()
        {
        }

        public AppResponseDto(T result) : this(result, true)
        {
        }

        public AppResponseDto(T result, bool succeed)
        {
            Result = result;
            Succeed = succeed;
        }
    }
}

File: CastIt.Android/lib/models/dtos/app_response_dto.dart
import 'package:json_annotation/json_annotation.dart';

import '../../common/utils/json_generic_converter.dart';
import 'empty_response_dto.dart';

part 'app_response_dto.g.dart';

@JsonSerializable()
class AppResponseDto<T> extends EmptyResponseDto {
  @JsonGenericConverter()
  @JsonKey(name: 'Result')
  T? result;

  AppResponseDto({
    this.result,
  });

  factory AppResponseDto.fromJson(Map<String, dynamic> json) => _$AppResponseDtoFromJson(json);
}
